cdt_list.go

```
@@ -294,10 +296,19 @@ func packCDTIfcVarParamsAsArray(packer BufferEx, opType int16, ctx []*CDTContext
 			}
 			size += n
 
-			if n, err = c.Value.pack(packer); err != nil {
-				return size + n, err
+			if c.Value != nil {
+				if n, err = c.Value.pack(packer); err != nil {
+					return size + n, err
+				}
+				size += n
+			} else if c.Expression != nil {
+				if n, err = c.Expression.pack(packer); err != nil {
+					return size + n, err
+				}
+				size += n
+			} else {
+				return size, newError(types.PARAMETER_ERROR, "CDTContext must have either a Value or an Expression")
```

should we also raise an error if both `Value` and `Expression` are present? If they are mutually exclusive, I think an error will be helpful here when both of them are present.